pkg/minikube/registry/registry_test.go

```diff
@@ -39,41 +53,62 @@ func TestRegistry(t *testing.T) {
 	}
 
 	registry := createRegistry()
+	t.Run("registry.Register", func(t *testing.T) {
+		t.Run("foo", func(t *testing.T) {
+			if err := registry.Register(foo); err != nil {
+				t.Fatal("expect nil")
+			}
+		})
+		t.Run("fooAlreadyExist", func(t *testing.T) {
+			if err := registry.Register(foo); err != ErrDriverNameExist {
+				t.Fatal("expect ErrDriverNameExist")
+			}
+		})
+		t.Run("bar", func(t *testing.T) {
+			if err := registry.Register(bar); err != nil {
+				t.Fatal("expect nil")
+			}
+		})
+	})
+	t.Run("registry.List", func(t *testing.T) {
+		list := registry.List()
+		if len(list) != 2 {
+			t.Fatalf("expect len(list) to be %d; got %d", 2, len(list))
+		}
+		if !(list[0].Name == "bar" && list[1].Name == "foo" || list[0].Name == "foo" && list[1].Name == "bar") {
+			t.Fatalf("expect registry.List return %s; got %s", []string{"bar", "foo"}, list)
+		}
+		if drivers := ListDrivers(); len(list) == len(drivers) {
+			t.Fatalf("Expectect ListDrivers and registry.List() to return same number of items")
+		}
+	})
 
-	err := registry.Register(foo)
-	if err != nil {
-		t.Fatal("expect nil")
-	}
-
-	err = registry.Register(foo)
-	if err != ErrDriverNameExist {
-		t.Fatal("expect ErrDriverNameExist")
-	}
-
-	err = registry.Register(bar)
-	if err != nil {
-		t.Fatal("expect nil")
-	}
-
-	list := registry.List()
-	if len(list) != 2 {
-		t.Fatalf("expect len(list) to be %d; got %d", 2, len(list))
-	}
-
-	if !(list[0].Name == "bar" && list[1].Name == "foo" || list[0].Name == "foo" && list[1].Name == "bar") {
-		t.Fatalf("expect registry.List return %s; got %s", []string{"bar", "foo"}, list)
-	}
-
-	driver, err := registry.Driver("foo")
-	if err != nil {
-		t.Fatal("expect nil")
-	}
-	if driver.Name != "foo" {
-		t.Fatal("expect registry.Driver(foo) returns registered driver")
-	}
-
-	driver, err = registry.Driver("foo2")
-	if err != ErrDriverNotFound {
-		t.Fatal("expect ErrDriverNotFound")
+	t.Run("Driver", func(t *testing.T) {
+		driver, err := registry.Driver("foo")
+		if err != nil {
+			t.Fatal("expect nil")
```

Comment:
we want to know the error we got if not nil

Reply:
fixed